lib/AST/USRGeneration.cpp

```diff
@@ -135,8 +135,10 @@ bool ide::printDeclUSR(const ValueDecl *D, raw_ostream &OS) {
   } else if (isa<TypeAliasDecl>(VD) || isa<AssociatedTypeDecl>(VD)) {
     Mangler.mangleContextOf(VD);
     Mangler.mangleDeclName(VD);
-  } else {
+  } else if (!VD->isInvalid()) {
     Mangler.mangleEntity(VD, /*uncurryingLevel=*/0);
```

Comment:
Same argument can be placed on `isInvalid`. Why cannot we make it disciplined and less random? Mangling invalid decls may cause unexpected results such the very crash this patch is trying to fix. Also, i think it is a valid assumption that mangling happens on valid declarations.